### DISCUSSION TITLE
Specify preprocessor definition for building dependency

### DISCUSSION THREAD

**Discussion by vid512** [mesonbuild/meson#11301] [Q&A]
What is the correct way to add preprocessor definition to only single dependency? In particular, I am trying to set CATCH_CONFIG_CONSOLE_WIDTH to 120 when building catch2.

**Comment by vid512** [comment]
I've found this in catch2 src/catch/meson.build:
```
conf_data = configuration_data()
conf_data.set('CATCH_CONFIG_DEFAULT_REPORTER', 'console')
conf_data.set('CATCH_CONFIG_CONSOLE_WIDTH', '80')

configure_file(
  input: 'catch_user_config.hpp.in',
  output: 'catch_user_config.hpp',
  format: 'cmake@',
  install_dir: get_option('includedir') / 'catch2',
  configuration: conf_data,
)
```

Does that mean meson build simply hardcodes this value and doesn't provide any mechanism to override it by user?

**Reply by eli-schwartz** [reply]
Looks like it. Maybe it would make sense to add a PR to upstream catch2 that makes it use meson_options.txt and read the user selected value using get_option().
